src/algebra/punit_instances.lean

```diff
@@ -69,27 +109,39 @@ instance : linear_ordered_cancel_add_comm_monoid punit :=
   decidable_lt := λ _ _, decidable.false,
   .. punit.canonically_ordered_add_monoid }
 
-instance (R : Type u) [semiring R] : module R punit := module.of_core $
-by refine
-{ smul := λ _ _, star,
-  .. punit.comm_ring, .. };
+instance : has_scalar R punit :=
+{ smul := λ _ _, star }
+
+@[simp] lemma smul_eq (r : R) : r • y = star := rfl
+
+instance : smul_comm_class R S punit := ⟨λ _ _ _, subsingleton.elim _ _⟩
+
+instance [has_scalar R S] : is_scalar_tower R S punit := ⟨λ _ _ _, subsingleton.elim _ _⟩
+
+instance [has_zero R] : smul_with_zero R punit :=
+by refine { ..punit.has_scalar, .. };
 intros; exact subsingleton.elim _ _
 
-@[simp] lemma zero_eq : (0 : punit) = star := rfl
-@[simp, to_additive] lemma one_eq : (1 : punit) = star := rfl
-@[simp] lemma add_eq : x + y = star := rfl
-@[simp, to_additive] lemma mul_eq : x * y = star := rfl
-@[simp, to_additive] lemma div_eq : x / y = star := rfl
-@[simp] lemma neg_eq : -x = star := rfl
-@[simp, to_additive] lemma inv_eq : x⁻¹ = star := rfl
-lemma smul_eq : x • y = star := rfl
-@[simp] lemma top_eq : (⊤ : punit) = star := rfl
-@[simp] lemma bot_eq : (⊥ : punit) = star := rfl
-@[simp] lemma sup_eq : x ⊔ y = star := rfl
-@[simp] lemma inf_eq : x ⊓ y = star := rfl
-@[simp] lemma Sup_eq : Sup s = star := rfl
-@[simp] lemma Inf_eq : Inf s = star := rfl
-@[simp] protected lemma le : x ≤ y := trivial
-@[simp] lemma not_lt : ¬(x < y) := not_false
+instance [monoid R] : mul_action R punit :=
+by refine { ..punit.has_scalar, .. };
+intros; exact subsingleton.elim _ _
+
+instance [monoid R] : distrib_mul_action R punit :=
+by refine { ..punit.mul_action, .. };
+intros; exact subsingleton.elim _ _
+
+instance [monoid R] : mul_distrib_mul_action R punit :=
+by refine { ..punit.mul_action, .. };
+intros; exact subsingleton.elim _ _
+
+/-! TODO: provide `mul_semiring_action R punit` -/
+-- importing it here currently causes timeouts elsewhere due to the import order changing
```

Comment:
This was causing horrible timeouts in files I don't care about, so in the interest of getting all the other isntances merged I'm going to skip this one.